### PR TITLE
Fullscreen support part 2

### PR DESCRIPTION
**Pull request by rolandkakonyi**
This PR adds support for full screen handling on iOS.

Additionally it implements cleaning up underlying components related to full screen handling once the player view gets deallocated.

### CODE REVIEW

File: android/src/main/java/com/bitmovin/player/reactnative/ui/FullscreenHandlerModule.kt [hunk 1]
```diff
@@ -32,7 +32,7 @@ class FullscreenHandlerModule(private val context: ReactApplicationContext) : Re
 
     fun requestEnterFullscreen(nativeId: NativeId) {
         context.catalystInstance.callFunction(
-            "bmFullscreenBridge-${nativeId}",
+            "FullscreenBridge-${nativeId}",
```
Comment (rolandkakonyi) [inline android/src/main/java/com/bitmovin/player/reactnative/ui/FullscreenHandlerModule.kt:35]: Removed prefix as we don't use it anywhere else.

File: example/ios/Podfile.lock [hunk 2]
```diff
@@ -503,7 +503,7 @@ SPEC CHECKSUMS:
   BitmovinPlayer: ae0fa04c43ea513d4fb2ae93fc2d5f2a271d92cc
```
Comment (rolandkakonyi) [inline example/ios/Podfile.lock:503]: Changes in this file are only due to minor dependency updates for iOS.

File: example/src/screens/BasicFullscreenHandling.tsx [hunk 3]
```diff
@@ -8,7 +8,6 @@ import {
   AudioSession,
```
Comment (rolandkakonyi) [inline example/src/screens/BasicFullscreenHandling.tsx:8]: I reworked the sample to make the player view actually full screen by hiding the navigation bar.

File: example/src/screens/BasicFullscreenHandling.tsx [hunk 4]
```diff
@@ -103,16 +97,19 @@ const styles = StyleSheet.create({
     flex: 1,
     alignItems: 'center',
     justifyContent: 'center',
-    backgroundColor: 'black',
+    backgroundColor: 'white',
+    padding: 20,
```
Comment (rolandkakonyi) [inline example/src/screens/BasicFullscreenHandling.tsx:101]: Added white background and a bit of padding to make it more visual when the player becomes fullscreen.

File: src/components/PlayerView/index.tsx [hunk 5]
```diff
@@ -103,6 +103,10 @@ export function PlayerView({
         );
       }
     }
+    return () => {
+      fullscreenBridge.current?.destroy();
+      fullscreenBridge.current = undefined;
+    };
```
Comment (rolandkakonyi) [inline src/components/PlayerView/index.tsx:109]: This was added to clean up fullscreen bridge once the player view gets deallocated.

File: src/ui/fullscreenhandlerbridge.ts [hunk 6]
```diff
@@ -11,16 +11,28 @@ const FullscreenHandlerModule = NativeModules.FullscreenHandlerModule;
...
+  /**
+   * Destroys the native FullscreenHandler
+   */
+  destroy() {
+    if (!this.isDestroyed) {
+      FullscreenHandlerModule.destroy(this.nativeId);
+      this.isDestroyed = true;
+    }
+  }
+
```
Comment (rolandkakonyi) [inline src/ui/fullscreenhandlerbridge.ts:35]: This was added to clean up fullscreen bridge once the player view gets deallocated.

File: src/ui/fullscreenhandlerbridge.ts [hunk 7]
```diff
@@ -11,16 +11,28 @@ const FullscreenHandlerModule = NativeModules.FullscreenHandlerModule;
...
   readonly nativeId: string;
   fullscreenHandler?: FullscreenHandler;
+  isDestroyed: boolean;
 
   constructor(nativeId?: string) {
     this.nativeId = nativeId ?? Uuid.generate();
+    this.isDestroyed = false;
     BatchedBridge.registerCallableModule(
-      `bmFullscreenBridge-${this.nativeId}`,
+      `FullscreenBridge-${this.nativeId}`,
```
Comment (rolandkakonyi) [inline src/ui/fullscreenhandlerbridge.ts:20]: Removed prefix as we don't use it anywhere else.

File: src/components/PlayerView/index.tsx [hunk 8]
```diff
@@ -103,7 +103,11 @@ export function PlayerView({
         );
       }
     }
-  }, [player, fullscreenBridge]);
```
Comment (rolandkakonyi) [inline src/components/PlayerView/index.tsx:106]: Having `fullscreenBridge` there prevents the player view being rendered.

File: android/src/main/java/com/bitmovin/player/reactnative/ui/FullscreenHandlerModule.kt [hunk 9]
```diff
@@ -43,7 +43,7 @@ class FullscreenHandlerModule(private val context: ReactApplicationContext) : Re
 
     fun requestExitFullscreen(nativeId: NativeId) {
         context.catalystInstance.callFunction(
-            "bmFullscreenBridge-${nativeId}",
+            "FullscreenBridge-${nativeId}",
```
[the remark this answers is not in the document]
Comment (strangesource) [reply android/src/main/java/com/bitmovin/player/reactnative/ui/FullscreenHandlerModule.kt:46]: Is the lower case to upper case change intentional?

**Review by strangesource — APPROVED** [review]
To my untrained eye this looks pretty good. 👍

File: android/src/main/java/com/bitmovin/player/reactnative/ui/FullscreenHandlerModule.kt (hunk 9 above)
[the remark this answers is not in the document]
Comment (rolandkakonyi) [reply android/src/main/java/com/bitmovin/player/reactnative/ui/FullscreenHandlerModule.kt:46]: Yes, all the `callFunction` String identifiers are starting with uppercase in other modules.

**Pull request merged**